### DISCUSSION TITLE
passing "default" to useStyles variantMaps

### DISCUSSION THREAD

**Discussion by jwfwessels** [jpudysz/react-native-unistyles#213] [General]
Hi,

I wanted to ask about this as I found the behavior a bit unexpected (I did double check the [documentation](https://reactnativeunistyles.vercel.app/reference/variants/#default-variant)).
We extensively use variants and breakpoints in our design system which we are building using Unistyles (thanks for a great library!)

But we have had to resort to a pattern like this all over the place because variantsMap doesn't accept default as a valid value.
``` ts
  const { styles } = useStyles(TextLinkStyles, {
    isVisited,
    isHovered,
    isActive,
    size: size === "default" ? undefined : size,   // <= would have expected to just pass through size
  });
  ```
Is it feasible to extend the API to allow for the string 'default'?

**Comment by jpudysz** [comment]
Hey @jwfwessels !

“Default” is a reserved word and means “no value for variant.” What you’re trying to achieve is to name your variant “default” and actually pass the name of the variant. It was intentionally excluded in the TypeScript type and supported in the core of Unistyles. Changing this mechanism would be a breaking change, and I’m against it.

You can simply name your variant differently, like “base” or “regular” and it will work as expected. I hope you understand!

**Reply by jwfwessels** [reply]
Yeah I get it. 

Im a bit stuck with default as im working with a large token library which uses the property name default all over the place to indicate the 'default' variant for properties and components. I guess I'll have to stick to my approach or monkey patch the types. 

Thanks.
